infra/build_specified_commit.py

```diff
@@ -152,6 +157,33 @@ def _build_image_with_retries(project_name):
   return result
 
 
+def get_required_post_checkout_steps(dockerfile_path):
+  """Get required post checkout steps (best effort)."""
+
+  checkout_pattern = re.compile(r'\s*RUN\s*(git|svn|hg)')
```

Comment:
nit: these 2 re.compile can be globals to create once.

Reply:
I think it's better to have them closer to where they're used to aid in comprehension, and constructing them will have a negligible cost.